### ISSUE TITLE
[Pay 2023-08-31] [$2000] Web - Contacts are being shown twice after creating a new group and a new workspace

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#21706]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open a new expensify account and sign in
2. Click on FAB > New group
3. Select a few members and create the group
4. Click on Settings > Workspace > New workspace
5. Go to workspace settings > Members
6. Invite all the members you have added to the group you created
7. Go to home and click on FAB > New group or click on the search icon at the top
8. Observe the contacts listed

## Expected Result:
One contact should be listed for one account

## Actual Result:
One account email is being shown twice on the new group or search contact list

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.31-2
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/93399543/d353d2dd-7684-4062-b13e-3618f890c254

https://github.com/Expensify/App/assets/93399543/e5f7b532-9b3c-49b8-ba82-9f25096d5cec

**Expensify/Expensify Issue URL:**
**Issue reported by:** @Nathan-Mulugeta
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1687535779695869

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01cf2a06b0a928050c</li>
        <li>Upwork Job ID: 1673974741833641984</li>
        <li>Last Price Increase: 2023-07-17</li>
<li>Automatic offers: </li>
<ul>
<li>aimane-chnaif | Reviewer | 26037525</li>
<li>StevenKKC | Contributor | 26037528</li>
<li>Nathan-Mulugeta | Reporter | 26037530</li>
</ul></li>
    </ul>
</details>

**Comment by s-alves10** [comment]
I don't think this is a bug. There exists separate reports for workspace members and group members for sure.

**Comment by conorpendergrast** [comment]
You don't even need to invite to the workspace to see this bug: when you enter the full email into the Invite field, the user shows up twice: 

<img width="440" alt="image" src="https://github.com/Expensify/App/assets/5802477/e7f0b1ff-bd34-4e3c-addd-47942fbffd77">

**Comment by conorpendergrast** [comment]
Regardless, I have reproduced this. This is indeed a bug, as users should not appear twice when you are creating a group DM (or inviting to a workspace).

<img width="468" alt="image" src="https://github.com/Expensify/App/assets/5802477/de2dd539-097b-445a-88ab-d6c3d588765f">

**Comment by conorpendergrast** [comment]
This appears to be reproducible anywhere that you can filter a contact lists, such as: 

| Assigning a task  | Second money | Starting a new 1:1 DM |  Splitting a bill | Requesting money
| ------------- | ------------- | ------------- | ------------- | ------------- |
| <img width="488" alt="image" src="https://github.com/Expensify/App/assets/5802477/07d6b63f-6a3d-4323-84fc-5b25130da179">  | <img width="413" alt="image" src="https://github.com/Expensify/App/assets/5802477/ec800deb-f528-459f-abf1-3bb357f634bf"> |  <img width="477" alt="image" src="https://github.com/Expensify/App/assets/5802477/32ed78c2-1cbf-4275-b49b-473cee6f9105"> | <img width="470" alt="image" src="https://github.com/Expensify/App/assets/5802477/a6d82993-cbe0-4036-8a7b-4d1d0d1932d7"> | <img width="373" alt="image" src="https://github.com/Expensify/App/assets/5802477/a84d1600-a036-4076-b6bf-1f413a387bab"> |

**Comment by niravkakadiya25** [comment]
https://expensify.slack.com/archives/C049HHMV9SM/p1686830123862429
This is fixed. think.
and this is a dupe.

**Comment by niravkakadiya25** [comment]
hi @aimane-chnaif 
can you click on the slack link? Where discussed the same things.

**Comment by conorpendergrast** [comment]
Oh I reproduced this 4 hours ago, on v1.3.33-4 (OSX desktop) so I'm pretty confident it's still an issue

**Comment by StevenKKC** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Contacts are being shown twice after creating a new group and a new workspace.

### What is the root cause of that problem?
When create a new group, `Report.navigateToAndOpenReport` is called.
https://github.com/Expensify/App/blob/3c339b72700ccaa6d6b2aadcdde39b2fe2febc3b/src/pages/NewChatPage.js#L205-L214

In `Report.navigateToAndOpenReport`, because this is a new group, will call `PersonalDetailsUtils.getAccountIDsByLogins`.
https://github.com/Expensify/App/blob/3c339b72700ccaa6d6b2aadcdde39b2fe2febc3b/src/libs/actions/Report.js#L498-L501

In `PersonalDetailsUtils.getAccountIDsByLogins`, because there are no associated personal detail, will generate an account ID for the users of a new group, and set to report's `participantAccountIDs` field.
https://github.com/Expensify/App/blob/3c339b72700ccaa6d6b2aadcdde39b2fe2febc3b/src/libs/PersonalDetailsUtils.js#L62-L64

These account IDs will be saved in `Onyx` by the following code.
https://github.com/Expensify/App/blob/3c339b72700ccaa6d6b2aadcdde39b2fe2febc3b/src/libs/actions/Report.js#L443-L458

After `OpenReport` API request success, server will return `report` information and `personalDetailsList` information in response, and `personalDetailsList` will contain correct user information.


These user information will be merged in `Onyx` data, so duplicated user data will be exist in `Onyx`, one with correct user id, and another with generated user id.
Therefore one account email is being shown twice on the new group or search contact list.
### What changes do you think we should make in order to solve the problem?

No need save new generated user information in `Onyx`, because server will return correct user information.
So we can remove these code in `Report.openReport`.
https://github.com/Expensify/App/blob/3c339b72700ccaa6d6b2aadcdde39b2fe2febc3b/src/libs/actions/Report.js#L443-L458

### What alternative solutions did you explore? (Optional)
None.

**Comment by conorpendergrast** [comment]
Currently waiting on more proposals, and for @aimane-chnaif to review [the proposal](https://github.com/Expensify/App/issues/21706#issuecomment-1611730571)

**Comment by conorpendergrast** [comment]
Interesting: I've tried to reproduce on Staging today, and couldn't get that reproduced. I'll re-test on Monday and will close if I still can't get it (and anyone else can try, too!)

**Comment by conorpendergrast** [comment]
Very weird, I'll re-test today!

**Comment by conorpendergrast** [comment]
ok, I still wasn't able to reproduce here. Here's what I did: 
1. Signed in to conor@expensifail.com
2. Created a new group DM with test+21706AA@conorpendergrast.com, test+21706BB@conorpendergrast.com, test+21706CC@conorpendergrast.com and test+21706DD@conorpendergrast.com (all four are new accounts that didn't exist before today)
3. Created a new workspace, and invited test+21706AA@conorpendergrast.com, test+21706BB@conorpendergrast.com, test+21706CC@conorpendergrast.com and test+21706DD@conorpendergrast.com
4. Click New Chat, and filtered by `test+21706AA@conorpendergrast.com` and `test+21706AA` and verified that the email was only listed a single time

Screenshots 1-3 (first row) and 4-6 (second row)

|   |  |  |
| ------------- | ------------- | ------------- |
| <img width="418" alt="image" src="https://github.com/Expensify/App/assets/5802477/ca918060-3de2-4fe1-9557-7c3cb5519cb6">  | <img width="972" alt="image" src="https://github.com/Expensify/App/assets/5802477/41670f61-94f6-433c-bb73-9b5e5527fe6e"> | <img width="405" alt="image" src="https://github.com/Expensify/App/assets/5802477/9feab9a7-c364-45b0-9258-40fd5cb2f6ee"> |
| <img width="408" alt="image" src="https://github.com/Expensify/App/assets/5802477/498f533a-a7cc-4589-9a19-41ee17b0c8ae"> | <img width="389" alt="image" src="https://github.com/Expensify/App/assets/5802477/8d1400b6-a1a7-4ab2-ad45-e5bba7af39c4"> | <img width="406" alt="image" src="https://github.com/Expensify/App/assets/5802477/9e58985c-02ed-4376-a58e-f8cc6696c545"> |

**Comment by conorpendergrast** [comment]
One difference from your reproduction steps, @aimane-chnaif , is that I didn't use a new account. Surely it can't be that critical to have a brand new account?

**Comment by Nathan-Mulugeta** [comment]
Okay @aimane-chnaif I will retest this again

**Comment by Nathan-Mulugeta** [comment]
I just retested this and I will attach a video down below. The steps are exactly the same as the OP, the thing I have noticed is that the account should be a new account. And I added 8 members in the group and added the same 8 members in the new workspace as well. This way I can reproduce it repeatedly. https://github.com/Expensify/App/assets/111440031/93d48ce4-08cc-45be-af1b-ecf837c710b4

**Comment by conorpendergrast** [comment]
Ok, fantastic. Agreed, thanks all

<img width="380" alt="image" src="https://github.com/Expensify/App/assets/5802477/87c5bdb1-3b6b-4788-a88c-1323808270bc">

**Comment by Beamanator** [comment]
FYI this is happening even when creating 1:1 DMs - are we handling that in this issue too or a different one?:

<img width="1029" alt="Screenshot 2023-07-13 at 11 58 59 AM" src="https://github.com/Expensify/App/assets/3885503/6b5893a7-d5cd-4967-a6a1-1d682aff565e">

**Comment by Beamanator** [comment]
@StevenKKC 

> No need save new generated user information in Onyx, because server will return correct user information.
So we can remove these code in Report.openReport.

Are you sure this will work in the offline case? I believe we do still need to have optimistic account data in Onyx for the offline case.

I think the solution will have to deal with removing the optimistic account data in `successData` and maybe even `failureData`

**Comment by Beamanator** [comment]
Yeah that's very odd 🤔 I haven't tried to reproduce the 8-person group DM

**Comment by StevenKKC** [comment]
@Beamanator I think my alternative solution would work fine in offline mode.

**Comment by conorpendergrast** [comment]
My reproduction was with a 4-person group DM, btw, not with the full 8 people. Bumping up the price here

**Comment by StevenKKC** [comment]
@conorpendergrast @Beamanator @aimane-chnaif 

BE responds `personalDetailsList` without `login` field, so these personal details will be filtered in [this line](https://github.com/Expensify/App/blob/9004c3968fcde7f46a473db871b4dd6ab2f3effc/src/pages/workspace/WorkspaceInvitePage.js#L150).
![Screenshot 2023-07-17 131733](https://github.com/Expensify/App/assets/131836113/c9518104-6e56-43ba-b7ce-776cd18302d4)
Therefore, new group chat with 8 members seems to does create duplicate contacts.

But in `Onyx`, there are duplicated personal details with different account id.
![Screenshot 2023-07-17 131607](https://github.com/Expensify/App/assets/131836113/817d3dd5-479d-46f8-bc2d-107833068496)

**Comment by StevenKKC** [comment]
@aimane-chnaif 
Only new user does not have `login` field.
And if it is expected, all pages which uses `login` field should be changed, including the invite new members page.
I think this is an another issue, and as you can see the screenshot, there are duplicated users with same `displayName` in `Onyx`.

**Comment by conorpendergrast** [comment]
Still looking for proposals here, I think

**Comment by Gonals** [comment]
There has been an active discussion going on regarding proposals this week, so I'm keeping it as external for a bit longer to see if we can nail down a solution in the next couple of days @aimane-chnaif, @conorpendergrast

**Comment by Gonals** [comment]
> Also we can't simply compare with `displayName` to avoid duplication because 3rd user can set 2nd user's email as first name, which is set as displayName.

I'm not sure I'm 100% following, but:

> Also we can't simply compare with displayName to avoid duplication because 3rd user can set 2nd user's email as first name, which is set as displayName.

This is possible, but it would be very rare, right? What if we prevent people from setting their name to any email that is not theirs?

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by conorpendergrast** [comment]
I've unassigned Bondy, as @Gonals is already assigned!

**Comment by bondydaa** [comment]
so I should assign @StevenKKC ?

**Comment by bondydaa** [comment]
![](https://media4.giphy.com/media/rOO3fE71nA9EY/giphy.gif?cid=ecf05e47fnktjsxqrqg6478bg0djygkak5hquo0ms63jpwn7&ep=v1_gifs_search&rid=giphy.gif&ct=g)

jk jk yeah I think ignoring the search case for now is fine.

**Comment by StevenKKC** [comment]
@aimane-chnaif PR https://github.com/Expensify/App/pull/24355 is ready for review.

**Comment by StevenKKC** [comment]
@conorpendergrast This was deployed on production 7 days back but the title wasn't updated. Could you please take a look at this?

**Comment by conorpendergrast** [comment]
I can!

**Comment by conorpendergrast** [comment]
I can't see anything that would explain why we wouldn't pay this today. Given that, I'm going to set this up to pay today (2023-08-31)

**Comment by conorpendergrast** [comment]
Payouts due:

- [x] Issue Reporter: $250 @Nathan-Mulugeta
- [x] Contributor: $2000 @StevenKKC 
- [x] Contributor+: $2000 @aimane-chnaif 

Eligible for 50% #urgency bonus? No

Upwork job is [here](https://www.upwork.com/ab/applicants/1673974741833641984/job-details).

**Comment by conorpendergrast** [comment]
All paid and regression test created https://github.com/Expensify/Expensify/issues/312805

We did it!

**Comment by conorpendergrast** [comment]
DAMMIT BONDY

**Comment by conorpendergrast** [comment]
Ok, bonuses sent. Now all done
